include/swift/SIL/SILBridgingImpl.h

```diff
@@ -1998,6 +2014,124 @@ BridgedWitnessTableEntry BridgedDefaultWitnessTable::getEntry(SwiftInt index) co
   return BridgedWitnessTableEntry::bridge(table->getEntries()[index]);
 }
 
+//===----------------------------------------------------------------------===//
+//                             BridgedKeyPathPattern
+//===----------------------------------------------------------------------===//
+
+BridgedKeyPathPatternComponentArray BridgedKeyPathPattern::components() const {
+  return {pattern->getComponents()};
+}
+
+BridgedStringRef BridgedKeyPathPattern::getObjCString() const {
+  return {pattern->getObjCString()};
+}
+
+//===----------------------------------------------------------------------===//
+//              BridgedKeyPathPatternComponent::ComputedPropertyId
+//===----------------------------------------------------------------------===//
+
+static_assert(sizeof(BridgedKeyPathPatternComponent::ComputedPropertyId) >=
+              sizeof(swift::KeyPathPatternComponent::ComputedPropertyId),
+              "BridgedKeyPathPatternComponent::ComputedPropertyId has wrong size");
+
+BridgedKeyPathPatternComponent::ComputedPropertyId::ComputedPropertyId(swift::KeyPathPatternComponent::ComputedPropertyId cpi) {
+  *reinterpret_cast<swift::KeyPathPatternComponent::ComputedPropertyId *>(&storage) = cpi;
+}
+
+swift::KeyPathPatternComponent::ComputedPropertyId BridgedKeyPathPatternComponent::ComputedPropertyId::unbridged() const {
+  return *reinterpret_cast<const swift::KeyPathPatternComponent::ComputedPropertyId *>(&storage);
+}
+
+//===----------------------------------------------------------------------===//
+//                         BridgedKeyPathPatternComponent
+//===----------------------------------------------------------------------===//
+
+static_assert(sizeof(BridgedKeyPathPatternComponent) >= sizeof(swift::KeyPathPatternComponent),
+              "BridgedKeyPathPatternComponent has wrong size");
+
+BridgedKeyPathPatternComponent::BridgedKeyPathPatternComponent(swift::KeyPathPatternComponent component) {
+  *reinterpret_cast<swift::KeyPathPatternComponent *>(&storage) = component;
+}
+
+swift::KeyPathPatternComponent BridgedKeyPathPatternComponent::unbridged() const {
+  return *reinterpret_cast<const swift::KeyPathPatternComponent *>(&storage);
+}
+
+BridgedKeyPathPatternComponent::Kind BridgedKeyPathPatternComponent::getKind() const {
```

Comment:
You could also just type cast between the enums and make sure the cases map to the same value with `static_assert`s. Like we do e.g. for `BridgedArgumentConvention` and many other enums.